doc/internals/index.rst

```diff
@@ -10,7 +10,6 @@ Here we document internal components of Arbor. These pages can be useful if you'
    :maxdepth: 2
 
    util
-   nmodl
    simd_api
-   sampling_api
+   extending_catalogues
```

Comment:
I think this needs to be moved to concepts/mechanisms. But maybe we can wait till the mechanism ABI work has been finalized.

Reply:
Exactly. I left it here, and once all is in one the ABI work it can move.